### PR TITLE
 first run

### PR DESCRIPTION
**Pull request by r-akers**

### CODE REVIEW

File: genisys/main.py [hunk 1]
```diff
@@ -66,3 +66,24 @@ def main():
...
+
+'''
+import needed modules,
+create new module with options/configs from other branches.
+'''
+def run():
+    import importlib
+
+    #netplan.py
+    netplan = importlib.import_module("genisys/modules/netplan.py")
```
Comment (gingrm02) [inline genisys/main.py:78]: Lets covert these to import statements as discussed before merging this

**Review by gingrm02 — CHANGES_REQUESTED** [review]

**Review by HenrithicusGreenson — APPROVED** [review]

**Review by patedm01 — APPROVED** [review]

**Review by BBergle — APPROVED** [review]

File: genisys/main.py [hunk 2]
```diff
@@ -66,3 +66,32 @@ def main():
...
+import needed modules,
+create new module with options/configs from other branches.
+'''
+def run():
+
+    #netplan.py
+    import modules.netplan as net
+
+    #get settings for functions
+    netplan = net.Netplan()
```
Comment (gingrm02) [inline genisys/main.py:80]: Needs config passed to the constructor.

File: genisys/main.py [hunk 3]
```diff
@@ -66,3 +66,32 @@ def main():
...
+    netLocation = net.install_location()
+    netgenerate = net.generate()
```
Comment (gingrm02) [inline genisys/main.py:82]: Generate should not have to be called manually on modules. Lets talk about the class inheritance architecture on the modules at some point.

File: genisys/main.py [hunk 4]
```diff
@@ -66,3 +66,32 @@ def main():
...
+
+    #base.py
+    import modules.base as base
```
Comment (gingrm02) [inline genisys/main.py:85]: Base should be treated as an abstract class. It is only meant to be inherited by the other modules, not instantiated directly.

File: genisys/main.py [hunk 5]
```diff
@@ -66,3 +66,32 @@ def main():
...
+
+    #get settings for functions
+    install = base.install(__path__)
+    validate = base.validate()
+
+    #configParser.py
+    import genisys.configParser as cp
```
Comment (gingrm02) [inline genisys/main.py:92]: The run function shouldn't need to make it's own instance of the configParser. The main function will parse the config based on certain command line options then pass that instance of the config to the run function.

**Review by gingrm02 — APPROVED** [review]
I'm approving this since we need it merged to address some of the comments. Lets talk through the system architecture soon to help understand some of my comments.

**Pull request closed without merging**